
# HG changeset patch
# User Sam Lantinga <slouken@libsdl.org>
# Date 1297589156 28800
# Node ID c2221eccce15bb75916c9cc66c57ef97a446b676
# Parent  8bc61743145169e54be65def344fff1001d2c5cb
Fixed bug with IMG_Init() return value.

Luke J Crook to sdl

The documentation for IMG_Init states: "You may call this function with a 0 to
retrieve whether support was built-in or not loaded yet."

What does this statement mean? I had assumed this to mean that calling IMG_Init
(0) will return the current status of JPG, PNG or TIF support. However looking
at the source, IMG_Init(0) will always return 0 as `flags` is bitwise ANDed
with IMG_INIT_JPG/IMG_INIT_PNG/IMG_INIT_TIF.

diff -r 8bc617431451 -r c2221eccce15 IMG.c
--- a/IMG.c	Tue Feb 08 09:32:16 2011 -0800
+++ b/IMG.c	Sun Feb 13 01:25:56 2011 -0800
@@ -90,7 +90,7 @@
 	}
 	initialized |= result;
 
-	return (result);
+	return (initialized);
 }
 
 void IMG_Quit()

